core/src/main/java/org/apache/calcite/rel/rules/JoinToMultiJoinRule.java

```diff
@@ -417,7 +417,14 @@ private void copyOuterJoinInfo(
    * @param nullGenerating true if the input is null generating
    * @return true if the input can be combined into a parent MultiJoin
    */
-  private boolean canCombine(RelNode input, boolean nullGenerating) {
+  private boolean canCombine(RelNode input, JoinRelType type, boolean nullGenerating) {
```

Comment:
Do we need `nullGenerating` since we can get it from `JoinRelType`?

Reply:
I don't think so in the case `input` is MJ but does not only contain full outer joins